staramr/detection/AMRDetection.py

```diff
@@ -177,14 +179,223 @@ def run_amr_detection(self, files, pid_threshold, plength_threshold_resfinder, p
             self._pointfinder_dataframe = self._create_pointfinder_dataframe(pointfinder_blast_map, pid_threshold,
                                                                              plength_threshold_pointfinder, report_all)
 
-        self._summary_dataframe = self._create_amr_summary(files, self._resfinder_dataframe,
+        self._summary_dataframe = self._create_amr_summary(files, self._resfinder_dataframe,self._quality_module_dataframe,
                                                            self._pointfinder_dataframe, self._plasmidfinder_dataframe, self._mlst_dataframe)
 
-        self._detailed_summary_dataframe = self._create_detailed_amr_summary(files, self._resfinder_dataframe,
+        self._detailed_summary_dataframe = self._create_detailed_amr_summary(files, self._resfinder_dataframe,self._quality_module_dataframe,
                                                                              self._pointfinder_dataframe,
                                                                              self._plasmidfinder_dataframe,
                                                                              self._mlst_dataframe)
 
+    def _create_quality_module_dataframe(self,files):
+        name_set=[]
+        for myFile in files:
+            name_set.append(path.splitext(path.basename(myFile))[0])
+        files_genome_lengths = self._get_genome_lengths(files)
+        files_genome_length_feedback = self._get_genome_length_feedback(files_genome_lengths)
```

Comment:
_get_genome_lengths and _get_genome_length_feedback should be combined into a single method.